pkg/mermaid/datamodeldiagram/datamodeldiagram.go

```diff
@@ -0,0 +1,155 @@
+			result += fmt.Sprintf("  %s %s\n", value.Type, cleanString(typeName))
+		case ref:
+			pair := externalLink{cleanString(appType), cleanString(value.Reference)}
+			if !externalLinksContain(*externalLinks, pair) {
+				*externalLinks = append(*externalLinks, pair)
+			}
+			result += fmt.Sprintf("  %s %s\n", cleanString(value.Reference), cleanString(typeName))
+		case "list", "set":
+			//TODO: check if list of lists is possible
+			if value.Items[0].Type == ref {
+				pair := externalLink{cleanString(appType), cleanString(value.Items[0].Reference)}
+				if !externalLinksContain(*externalLinks, pair) {
+					*externalLinks = append(*externalLinks, pair)
+				}
+				result += fmt.Sprintf("  List<%s> %s\n", cleanString(value.Items[0].Reference), cleanString(typeName))
+			} else {
+				result += fmt.Sprintf("  List<%s> %s\n", cleanString(value.Items[0].Type), cleanString(typeName))
+			}
+		default:
+			panic("Unrecognised type")
+		}
+	}
+	return result
+}
+
+//printMap prints appropriate mermaid code for values inside a map
+func printMap(properties map[string]*syslwrapper.Type,
+	appType string, primaryKey string, externalLinks *[]externalLink) string {
+	var result string
+	var type1, type2, name, name1, name2 string
+	name = appType
+	name1 = primaryKey
+	//TODO: sort elements in map for uniformity
+	for typeName, value := range properties {
+		if typeName == primaryKey {
+			type1 = value.Type
+		} else {
+			name2 = typeName
+			type2 = value.Type
+		}
+		switch value.Type {
+		case "int", "bool", "float", "decimal", "string", "string_8", "bytes", "date", "datetime", "xml", "uuid":
+			break
+		case ref:
+			pair := externalLink{cleanString(appType), cleanString(value.Reference)}
+			if !externalLinksContain(*externalLinks, pair) {
+				*externalLinks = append(*externalLinks, pair)
+			}
+		case "list", "set":
+			//TODO: check if list of lists is possible
+			if value.Items[0].Type == ref {
+				pair := externalLink{cleanString(appType), cleanString(value.Items[0].Reference)}
+				if !externalLinksContain(*externalLinks, pair) {
+					*externalLinks = append(*externalLinks, pair)
+				}
+			}
+		default:
+			panic("Unrecognised type")
+		}
+	}
+	result += fmt.Sprintf("  Map<%s, %s> %s<%s, %s>\n", type1, type2, name, name1, name2)
+	return result
+}
+
+//printEnum prints appropriate mermaid code for values inside an enum
+func printEnum(enum map[int64]string) string {
+	var result string
+	//TODO: sort keys in map for uniformity
+	for key, value := range enum {
+		result += fmt.Sprintf("  %s %d\n", value, key)
+	}
+	return result
+}
+
+//externalLinksContain checks if a connection between two data types exists or not
+func externalLinksContain(f []externalLink, s externalLink) bool {
+	for _, a := range f {
+		if a == s {
+			return true
+		}
+	}
+	return false
+}
+
+//cleanString replaces certain characters in the string suitable for mermaid
+func cleanString(temp string) string {
```

Comment:
We should probably document this for the CLI somewhere

Reply:
Alright cool

Reply:
I think once you're ready to release mermaid "publicly" you can document it in `sysl/docs/website/content/features/diagrams.md`

Reply:
Awesome!